Dependency injection

AntonKorenev opened the pull request:
This is a first version which was created tomorrow. I don't use autowiring now because of simple project structure but, if it's necessary, making changes won't be complex.
The logic of making ConnectorCSV and Mapper 'prototype' instead of 'singleton' is possible implementation multi threaded architecture in future builds. This classes aren't tied to service, DB or other structures without multi access support(write case) so that it is possible to create more than one object of them to maintain their functional routine. Actually, I plan them to make a copy of  incoming file  instead of changing it and that's why I think, the several their objects can exists simultaneously.

AntonKorenev committed:
to test branch switching in idea

AntonKorenev committed:
New:
1) Initial project structure were added. Dependencies between were established using xml DI way.
2) Gradle now automatically imports Appache common-logging lib and spring libs.
3) New gradle task for launching application and application tests. Now Gradle automatically opens report page in default browser.
4) Initial comments for javadoc.

Fix:
1) JUnit assert, which prevents to run showTest task in gradle was caught by standard GradleException instead of java specific one.

alexdik commented:
IDE files should not be present in repository, project must be IDE unaware.

alexdik commented on src/test/java/ProcessorTest.java:
```diff
@@ -0,0 +1,47 @@
+     * Method for testing whether Processor has access to Dashboard or not
+     */
+    @Test(timeout = 100)
+    public void testAccessToDashboard(){
```
Test name should describe what behaviour is expected, it could look like processorHasAccessToDashboard and check that processor has access to dashboard. Comments are required only if you do something not obvious or dirty, in all other cases it is better to have a self explanatory code.

alexdik commented:
Please implement component by component, isolated from each other (when possible). So I would expect single mapper class implementation complete along with unit tests per one commit. No need to update everything, but please consider it for future commits.

alexdik commented on src/main/java/com/company/SpringApplication/Publishers/PublisherDashboard.java:
```diff
@@ -0,0 +1,17 @@
+package com.company.SpringApplication.Publishers;
```
package names should be lower case, you can check java code conventions for code formatting

alexdik commented on src/main/java/com/company/SpringApplication/Processors/Processor.java:
```diff
@@ -0,0 +1,100 @@
+     * @param publisherUpstream Reference to {@link PublisherUpstream} class. It connects Processor with
+     *                          Upstream service.
+     */
+    public Processor(PublisherDashboard publisherDashboard,
```
no need to write comments until it is obvious

AntonKorenev committed:
New:
1) Temporary implementation of ConnectorCSV class (still without servlet part) and its junit test.

Changed:
2) spring-test dependency was added to build.gradle

AntonKorenev committed:
Just removing unncecessary IDE files from repository

AntonKorenev committed:
Just a small changes to dataArray in ConnectorCSV to simplify file converting

AntonKorenev committed:
New:
1) MapperXML implementation and it's unit test

Difficulties: idea and gradle from time to time works improper, that's why I was unable to run the last test

alexdik commented:
Before implementing mappers please consider to define data structure, it should be implemented as value object and eventually reused throughout application.
